agent/app/agent.go

```diff
@@ -108,6 +110,26 @@ func newAgent(
 		return nil, err
 	}
 
+	var metadataManager containermetadata.MetadataManager
+	if cfg.ContainerMetadataEnabled {
+		// Get the oldest Docker Client version with up to date inspect API
```

Comment:
Right now I am cautious about making changes here because I have encountered in the past import cycles and/or "interface is not implemented" errors when dealing with this section of code. I will see what I can do here.